jrqs-app/src/java/ua/dp/isd/jrqs/RequirementsService.java

```diff
@@ -27,22 +27,40 @@
 package ua.dp.isd.jrqs;
 
 import java.util.List;
+import ua.dp.isd.jrqs.ESearchBy.SearchBy;
 
 public class RequirementsService {
-    
-    public List<Requirement> getRequirementsByRowId(String RequirementRowIds) {
-        return new RequirementsDAO().getRequirementsByRowIds(RequirementRowIds);
-    }
-    
-    public List<Requirement> getRequirementsByRequirementNumbers(String RequirementNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByRequirementNumbers(RequirementNumbers, LimitBySource);
+
+    private final String rawParamsSearchBy;
+    private final String rawParamsUserInput;
+    private final String rawParamsLimitBySource;
+
+    public RequirementsService(String by, String value, String only) {
+        rawParamsSearchBy = by;
+        rawParamsUserInput = value;
+        rawParamsLimitBySource = only;
     }
-    
-    public List<Requirement> getRequirementsByTmsTaskNumbers(String TmsTaskNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByTmsTaskNumbers(TmsTaskNumbers, LimitBySource);
+
+    public List<Requirement> getRequirements() {
+        if (rawParamsSearchBy == null || rawParamsSearchBy.equals("")) {
+            throw new IllegalArgumentException("Invalid request: by parameter is missing in the request.");

```

Comment:
Is it really possible to have these fields as nulls? I mean, if they are never should be nulls, don't make any additional checks of this condition. Check them when the instance of the class is instantiated. In all other places, this check will not any clarity: you still will not be able to track the root of the problem by throwing this exception. Less exceptions → less assumptions.